uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -285,7 +291,14 @@ define([
             initAbsenceType();
             initStatus();
             initContact();
-            initDates();
+
+            //the promise will set the day types so that initial value is set properly
+            return initDates();
+          })
+          .then( function () {
+            if(vm.mode == 'edit') {
+              initialLeaveRequest = _.cloneDeep(vm.leaveRequest.attributes());
```

Comment:
You don't need to clone the result of `attributes()`, given that those attributes are stored in a brand new object
```js
attributes: function () {
    return _.transform(this, function (result, __, key) {
        !_.isFunction(this[key]) && (result[key] = this[key]);
    }, Object.create(null), this);
},
```